### ISSUE TITLE
spanner: segfault when credentials could not be found

### ISSUE THREAD

**Issue by kdeberk** [googleapis/google-cloud-go#1905]
**Client**

Spanner

**Environment**

Mac OS X Catalina

**Code**

e.g.
```go
package main

import (
	"context"
	"log"
	"cloud.google.com/go/spanner"
)

func main() {
	client, err := spanner.NewClientWithConfig(
		context.Background(),
		"projects/some-project/instances/some-instance/databases/some-database",
		spanner.ClientConfig{NumChannels: 1},
	)
	if err != nil {
		log.Fatalf("Could not create client: %v", err)
	}
	stmt := spanner.NewStatement("SELECT * FROM some-table LIMIT 1")
	client.Single().Query(context.Background(), stmt)
}
```

**Expected behavior**
Not crashing

**Actual behavior**
Crashing.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x14d550d]

goroutine 1 [running]:
cloud.google.com/go/spanner.NewClientWithConfig(0x174ccc0, 0xc0000b2000, 0x1690cdf, 0x45, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/kevindeberk/go/pkg/mod/cloud.google.com/go/spanner@v1.4.0/client.go:240 +0xcdd
main.main()
        /Users/kevindeberk/code/test/foo/spanner.go:16 +0xa2
```

Using `dlv` shows that an error is being ignored on client.go:239.

**Comment by kdeberk** [comment]
PR: https://code-review.googlesource.com/c/gocloud/+/54510

**Comment by codyoss** [comment]
Thank you for finding this, and for the contribution.

**Comment by codyoss** [comment]
Fixed with 90c5e7c8c8ef8adec7489e14bf5d8ef4b623f23e
